models/verifier/face_verifier.py

```diff
@@ -101,29 +137,41 @@ def verify(self, im1, im2, threshold=0.5, with_detection=False, return_distance=
         else:
             return is_same_person
     
-    def extract_embeddings(self, im, with_detection=False, return_face=False):
+    def extract_embeddings(self, im, with_detection=False, with_alignment=False, return_face=False):
+        """Extract latent embeddings of the input image
+
+        # Arguments
+            im: Numpy array, the input image.
+            with_detection: Boolean, whether to apply face detetion before feature extraction.
+            with_alignment: Boolean, whether to apply face alignment before feature extraction.
+            return_face: Boolean, whether to return the preprocessed face (input of feature extractor).
+
+        # Returns
+            embeddings: Numpy array, latent embeddings of input image.
+            face: Numpy array, preprocessed face as the input of feature extractor.
+        """
         if with_detection:
             try:
-                if self.extractor_type == "facenet":
-                    faces = self.detector.detect_face(im, with_landmarks=False)
-                elif self.extractor_type == "insightface":
+                if with_alignment:
                     faces, landmarks = self.detector.detect_face(im, with_landmarks=True)
                     landmarks = [self.detector.convert_landmarks_68_to_5(l) for l in landmarks]
+                else:
+                    faces = self.detector.detect_face(im, with_landmarks=False)
             except:
                 raise NameError("Error occured duaring face detection. \
-                Please check if face detector has been set through set_detector().")
+                Please check if face detector has been set through set_detector() and there are available GPUs.")
```

Comment:
The FAN error caused by not using GPU instance might be solve by re-converting FAN model in channel_last ordering using [pytorch2keras](https://github.com/nerox8664/pytorch2keras) package.